packages/google_maps_flutter/google_maps_flutter/ios/Classes/GoogleMapMarkerController.m

```diff
@@ -5,151 +5,294 @@
+  if (!controller) {
+    return;
+  }
+  [self.methodChannel
+      invokeMethod:@"marker#onDrag"
+         arguments:@{
+           @"markerId" : identifier,
+           @"position" : [FLTGoogleMapJsonConversions getJsonFromLocation:coordinate]
+         }];
+}
+- (void)didEndDraggingMarkerWithIdentifier:(NSString *)identifier
+                                coordinate:(CLLocationCoordinate2D)coordinate {
+  FLTGoogleMapMarkerController *controller = self.markerIdentifierToController[identifier];
+  if (!controller) {
+    return;
+  }
+  [self.methodChannel
+      invokeMethod:@"marker#onDragEnd"
+         arguments:@{
+           @"markerId" : identifier,
+           @"position" : [FLTGoogleMapJsonConversions getJsonFromLocation:coordinate]
+         }];
+}
+- (void)didTapInfoWindowOfMarkerWithIdentifier:(NSString *)identifier {
+  if (identifier && self.markerIdentifierToController[identifier]) {
+    [self.methodChannel invokeMethod:@"infoWindow#onTap" arguments:@{@"markerId" : identifier}];
+  }
+}
+- (void)showMarkerInfoWindowWithIdentifier:(NSString *)identifier result:(FlutterResult)result {
+  FLTGoogleMapMarkerController *controller = self.markerIdentifierToController[identifier];
+  if (controller) {
+    [controller showInfoWindow];
+    result(nil);
+  } else {
+    result([FlutterError errorWithCode:@"Invalid markerId"
+                               message:@"showInfoWindow called with invalid markerId"
+                               details:nil]);
+  }
+}
+- (void)hideMarkerInfoWindowWithIdentifier:(NSString *)identifier result:(FlutterResult)result {
+  FLTGoogleMapMarkerController *controller = self.markerIdentifierToController[identifier];
+  if (controller) {
+    [controller hideInfoWindow];
+    result(nil);
+  } else {
+    result([FlutterError errorWithCode:@"Invalid markerId"
+                               message:@"hideInfoWindow called with invalid markerId"
+                               details:nil]);
+  }
+}
+- (void)isInfoWindowShownForMarkerWithIdentifier:(NSString *)identifier
+                                          result:(FlutterResult)result {
+  FLTGoogleMapMarkerController *controller = self.markerIdentifierToController[identifier];
+  if (controller) {
+    result(@([controller isInfoWindowShown]));
+  } else {
+    result([FlutterError errorWithCode:@"Invalid markerId"
+                               message:@"isInfoWindowShown called with invalid markerId"
+                               details:nil]);
+  }
+}
 
-static CGPoint ToPoint(NSArray *data) { return [FLTGoogleMapJsonConversions toPoint:data]; }
++ (CLLocationCoordinate2D)getPosition:(NSDictionary *)marker {
+  NSArray *position = marker[@"position"];
+  return [FLTGoogleMapJsonConversions getLocationFromData:position];
+}
 
-static NSArray *PositionToJson(CLLocationCoordinate2D data) {
-  return [FLTGoogleMapJsonConversions positionToJson:data];
++ (NSString *)getMarkerIdentifier:(NSDictionary *)marker {
```

Comment:
`identifierForMarker:`

Or remove it, because the code is shorter than the wrapper...